File: PersonnelDepartment/Domain/Employees/EmployeeBlank.cs
namespace PersonnelDepartment.Domain.Employees;

public class EmployeeBlank
{
    public Guid? Id { get; set; }
    public Guid? PostId { get; set; }
    public String? Name { get; set; }
    public String? Surname { get; set; }
    public String? Partronymic { get; set; }
    public String? PhoneNumber { get; set; }
    public String? Email { get; set; }
    public String? Inn { get; set; }
    public String? Snils { get; set; }
    public Int32? PassportSeries { get; set; }
    public Int64? PassportNumber { get; set; }
    public DateTime? BirthDay { get; set; }
    public Boolean IsDismissed { get; set; }
}

File: PersonnelDepartment/ClientApp/src/domain/employees/employeeBlank.ts
import { Employee } from "./employee";

export class EmployeeBlank {
    constructor(
        public id: string | null,
        public departmentId: string | null,
        public postId: string | null,
        public name: string | null,
        public surname: string | null,
        public partronymic: string | null,
        public phoneNumber: string | null,
        public email: string | null,
        public inn: string | null,
        public snils: string | null,
        public passportSeries: number | null,
        public passportNumber: number | null,
        public birthDay: Date | null,
        public isDismissed: boolean
    ) { }

    public static empty(): EmployeeBlank {
        return new EmployeeBlank(null, null, null, null, null, null, null, null, null, null, null, null, null, false);
    }

    public static fromEmployee(employee: Employee): EmployeeBlank {
        return new EmployeeBlank(
            employee.id, employee.departmentId, employee.postId, employee.name, employee.surname,
            employee.partronymic, employee.phoneNumber, employee.email, employee.inn, employee.snils,
            employee.passportSeries, employee.passportNumber, employee.birthDay, employee.isDismissed
        )
    }
}